lib/evaluate/type.cc

```diff
@@ -163,10 +199,8 @@ std::optional<DynamicType> DynamicType::From(
   } else if (const auto *derived{type.AsDerived()}) {
     return DynamicType{
         *derived, type.category() == semantics::DeclTypeSpec::ClassDerived};
-  } else if (type.category() == semantics::DeclTypeSpec::ClassStar) {
-    return DynamicType::UnlimitedPolymorphic();
   } else {
-    // Assumed-type dummy arguments (TYPE(*)) do not have dynamic types.
+    return DynamicType::UnlimitedPolymorphic();
```

Comment:
We need to have a dynamic type for `TYPE(*)` when determining whether two procedures are distinguishable.

Reply:
On the way!  Will get it into the PR I'm about to merge.